BuildingRefrigerator/BuildingRefrigerator/refrigerator.cs

```diff
@@ -0,0 +1,140 @@
+﻿namespace BuildingRefrigerator;
+
+public class refrigerator : IComparable<refrigerator?>
+{
+    static private int _uniqueIdentifierRefrigerator = 0;
+    private List<shelf?>? _shelves;
+    private string? _model;
+    private int _maximumShelves;
+    public refrigerator(string? model, string color, int maximumShelves, List<shelf?>? shelves = null)
+    {
+        Id = ++_uniqueIdentifierRefrigerator;
+        Model = model;
+        Color = color;
+        _maximumShelves = maximumShelves;
+        Shelves = shelves;
+    }
+    public int Id { get; }
+    public string? Model
+    {
+        get=> _model;
+        set
+        {
+            if (value==null)
+                throw new Exception("A refrigerator model is missing. Cannot create a fridge");
+            _model = value;
+        }
+    }
+    public string Color { get; set; }
+    public List<shelf?>? Shelves
+    {
+        get => _shelves;
+        set
+        {
+            if (value is not null)
+            {
+                _shelves = value;
+            }
+            else
+            {
+                _shelves = new List<shelf?>();
+                for (int i = 1; i <= _maximumShelves; i++)
+                    _shelves.Add(new shelf(i));
```

Comment:
this is a very confusing interface, should be constructor logic.
also this allows to bypass the limit of shelfs.